controllers/PollController.ts

```diff
@@ -59,6 +67,79 @@ export default class PollController implements PollControllerI {
             .then((poll: Poll) => res.json(poll));
     }
 
+    /**
+     * Retrieves all polls from the database and returns an array of polls.
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the tuit objects
+     */
+    getAllPolls = (req: Request, res: Response) => {
+        PollController.pollDao.getAllPolls()
+            .then((poll: Poll[]) => res.json(poll));
+    }
+
+    /**
+     * Retrieves a poll from the database by id and returns a poll.
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the tuit objects
+     */
+    getPollById = (req: Request, res: Response) => {
+        PollController.pollDao.getPollById(req.params.pid)
+            .then((poll: Poll) => res.json(poll));
+    }
+
+
+    /**
+     * Deletes a poll if the delete request is from the creator. Else ignores the request
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the status of the delete operation
+     */
+    deletePoll = async (req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+
+        const poll = await PollController.pollDao.getPollById(req.params.pid);
+        if(poll != null) {
+            if (poll.createdBy.toString() === userId.toString()) {
+                PollController.pollDao.deletePoll(req.params.pid)
+                    .then((status) => res.json(status));
+            }
+            else {
+                res.json("Only creator can delete poll")
+            }
+        }
+
+        else {
+            res.sendStatus(404);
+        }
+
+    }
+
+
+    /**
+     * Close a poll if the request is from the creator. Else ignore the request
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents a response to client with the status of close operation
+     */
+    closePoll = async(req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+
+        const poll = await PollController.pollDao.getPollById(req.params.pid);
+        if(poll != null) {
```

Comment:
Might want to check if this would work for invalid pollId

Reply:
Performed testing in postman, it is working fine